exporter/rabbitmqexporter/integration_test.go

```diff
@@ -0,0 +1,162 @@
+
+			// Create and start rabbitmqexporter
+			factory := NewFactory()
+			cfg := factory.CreateDefaultConfig().(*Config)
+			cfg.Connection.Endpoint = endpoint
+			cfg.Connection.VHost = vhost
+			cfg.Connection.Auth = AuthConfig{Plain: PlainAuth{Username: username, Password: password}}
+			exporter, err := factory.CreateLogsExporter(context.Background(), exportertest.NewNopCreateSettings(), cfg)
+			require.NoError(t, err)
+			err = exporter.Start(context.Background(), componenttest.NewNopHost())
+			require.NoError(t, err)
+			defer func() {
+				err = exporter.Shutdown(context.Background())
+				require.NoError(t, err)
+			}()
+
+			// Export and verify data is consumed
+			logs := testdata.GenerateLogsOneLogRecord()
+			err = exporter.ConsumeLogs(context.Background(), logs)
+			require.NoError(t, err)
+			consumed := <-consumer
+			require.NotNil(t, consumed.Body)
+
+			// Stop the container before exporting the next logs to simulate a network issue
+			err = channel.Close()
+			require.NoError(t, err)
+			err = connection.Close()
+			require.NoError(t, err)
+			stopTimeout := time.Second * 5
+			err = container.Stop(context.Background(), &stopTimeout)
+			require.NoError(t, err)
+			logs = testdata.GenerateLogsOneLogRecord()
+			err = exporter.ConsumeLogs(context.Background(), logs)
+			require.Error(t, err)
+
+			// Restart container to simulate network issue recovery
+			err = container.Start(context.Background())
+			require.NoError(t, err)
+			connection, channel, consumer = setupQueueConsumer(t, logsRoutingKey, endpoint)
+			defer func() {
+				channel.Close()
+				connection.Close()
+			}()
+
+			logs = testdata.GenerateLogsOneLogRecord()
+			err = exporter.ConsumeLogs(context.Background(), logs)
+			require.NoError(t, err)
+			consumed = <-consumer
+			require.NotNil(t, consumed.Body)
+		})
+	}
+}
+
+func setupQueueConsumer(t *testing.T, queueName string, endpoint string) (*amqp.Connection, *amqp.Channel, <-chan amqp.Delivery) {
+	connection, err := amqp.DialConfig(endpoint, amqp.Config{
+		SASL: []amqp.Authentication{
+			&amqp.PlainAuth{
+				Username: username,
+				Password: password,
+			},
+		},
+		Vhost: vhost,
+	})
+	require.NoError(t, err)
+
+	channel, err := connection.Channel()
+	require.NoError(t, err)
+
+	queue, err := channel.QueueDeclare(queueName, false, false, false, false, nil)
+	require.NoError(t, err)
+
+	consumer, err := channel.Consume(queue.Name, "testconsumer", true, false, false, false, nil)
+	require.NoError(t, err)
+
+	return connection, channel, consumer
+}
+
+func getContainer(t *testing.T, req testcontainers.ContainerRequest) testcontainers.Container {
+	require.NoError(t, req.Validate())
+	container, err := testcontainers.GenericContainer(
+		context.Background(),
+		testcontainers.GenericContainerRequest{
+			ContainerRequest: req,
+			Started:          true,
+		})
+	require.NoError(t, err)
+
+	err = container.Start(context.Background())
```

Comment:
Is the container guaranteed to be up and running and ready when `Start` returns? (I couldn't tell from `testcontainer` docs) If not, we may need to add some kind of check to ensure it's ready by the time we're using it.

Reply:
I added another waiting strategy for good measure just in case there's some race condition until rabbitmq's ready https://golang.testcontainers.org/features/wait/introduction/